airbyte-integrations/connectors/source-jira/source_jira/streams.py

```diff
@@ -137,6 +172,12 @@ def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
             self.logger.warning(f"Stream `{self.name}`. An error occurred, details: {errors}. Skipping for now. {custom_error}")
 
 
+class FullRefreshJiraStream(JiraStream):
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
+        yield from super().read_records(**kwargs)
+        self.state = FULL_REFRESH_COMPLETE_STATE
```

Comment:
hey @darynaishchenko 

Gonna pick this discussion up here since this is presumably where you added the fix to stop the infinite loop. What I'm trying to figure out here is whether there is a bug in the RFR code that is causing the infinite loop or that Jira is configured in a way where we need this fix. 

Right now I feel like this fix you have here would work and it addressed the issue blocking the merge. However, it feels like a workaround to a CDK bug. Ideally we shouldn't have to fix this in the connector

- For RFR streams, we should automatically get the assignment of `self.state` to `FULL_REFRESH_COMPLETE_STATE` within the default implementation of `HttpStream.read_records()` when we finish reading a page [here](https://github.com/airbytehq/airbyte/blob/5056e6782639b05743dc77a9ee731e5f583bae86/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py#L408)
- For regular incremental or full refresh streams, those use checkpoint readers (IncrementalCheckpointReader / FullRefreshCheckpointReader) that don't rely on `FULL_REFRESH_COMPLETE_STATE` to stop iteration.

I would be okay if we merged this if we need to get this fix out, but I am going to try to debug on your branch for the `issue_comments` stream to see if I can see what's wrong. I'm okay w/ the merge, but let's add a comment that this is a temporary workaround. I'll keep everyone posted as I know more, but I suspect that we're assigning the wrong checkpoint reader

Reply:
Thanks Brian! I'll add comment about this workaround and merge these changes with temporary workaround. I'll run regression tests to be sure that it works fine before merge.

Reply:
comment added.
Let me know when you find something.

> but I am going to try to debug on your branch for the issue_comments stream to see if I can see what's wrong.